Wedge bond from atrop error

tadhurst-cdd opened the pull request:
<!--
Thanks for contributing a pull request! 
-->
#### Reference Issue
Lost atropiomser wedging when an neighboring bond already has a direction related to a double bond on it.


#### What does this implement/fix? Explain your changes.
Now that bond is simply not considered for the wedge to define the atropisomer, and another bond is chosen.  Previously, the search for a bond to wedge was terminated

#### Any other comments?

greglandrum commented on Code/GraphMol/FileParsers/file_parsers_catch.cpp:
```diff
@@ -6010,10 +6010,10 @@ TEST_CASE("MaeWriter basic testing", "[mae][MaeWriter][writer]") {
-    std::string_view atomBlock(&mae[atomBlockStart],
-                               bondBlockStart - atomBlockStart);
-    std::string_view bondBlock(&mae[bondBlockStart]);
+    std::string ctBlock(&mae[ctBlockStart], atomBlockStart - ctBlockStart);
```
why are these `string_view` -> `string` changes here?

tadhurst-cdd replied:
You are right  - these changes have nothing to do with the bug or the fix.   I needed to do this to get the system to build.

When I remove the changes I get:
[build] /usr/bin/ld: CMakeFiles/fileParsersCatchTest.dir/file_parsers_catch.cpp.o: in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Catch::Detail::stringify<std::basic_string_view<char, std::char_traits<char> > >(std::basic_string_view<char, std::char_traits<char> > const&)':
[build] /usr/local/include/catch2/catch_tostring.hpp:150: undefined reference to `Catch::StringMaker<std::basic_string_view<char, std::char_traits<char> >, void>::convert[abi:cxx11](std::basic_string_view<char, std::char_traits<char> >)'

I am using g++ (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
and clang version 10.0.0-4ubuntu1 .

Anyway, I have removed those changed and checked it back in.

Any suggestions on the build?


Suggestion

greglandrum replied:
> Any suggestions on the build?

This could be because you are still using an older version of cmake (or a version installed the way the RDKit used to do it before #6898).

You might try looking to see if there's a `catch` directory in `$RDBASE/External/catch` and/or `build/External/catch` and, if so, removing those directories, then re-running cmake. This will complain about not finding catch2 and then grab a new copy and build it using the proper build flags.

greglandrum reviewed (changes requested):
The changes to `file_parsers_catch.cpp` aren't connected to this change and I don't think I agree with them anyway, but otherwise this looks fine.

Please explain or revert those changes

greglandrum reviewed (approved):
LGTM

tadhurst-cdd commented:
I updated cmake, and I also removed catch2 libs and includes.  the cmake
command did not re-load catch2, but generated an error saying it could not
find catch2.
With the catch2 back in place, I still get the error about string_view.
 Any other ideas?

How do YOU load Catch2?

thanks, tad

On Wed, Apr 3, 2024 at 7:52 AM Greg Landrum ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Code/GraphMol/FileParsers/file_parsers_catch.cpp
> <https://github.com/rdkit/rdkit/pull/7321#discussion_r1549678842>:
>
> > @@ -6010,10 +6010,10 @@ TEST_CASE("MaeWriter basic testing", "[mae][MaeWriter][writer]") {
>      auto bondBlockStart = mae.find("m_bond[6]");
>      REQUIRE(bondBlockStart != std::string::npos);
>
> -    std::string_view ctBlock(&mae[ctBlockStart], atomBlockStart - ctBlockStart);
> -    std::string_view atomBlock(&mae[atomBlockStart],
> -                               bondBlockStart - atomBlockStart);
> -    std::string_view bondBlock(&mae[bondBlockStart]);
> +    std::string ctBlock(&mae[ctBlockStart], atomBlockStart - ctBlockStart);
>
> Any suggestions on the build?
>
> This could be because you are still using an older version of cmake (or a
> version installed the way the RDKit used to do it before #6898
> <https://github.com/rdkit/rdkit/pull/6898>).
>
> You might try looking to see if there's a catch directory in
> $RDBASE/External/catch and/or build/External/catch and, if so, removing
> those directories, then re-running cmake. This will complain about not
> finding catch2 and then grab a new copy and build it using the proper build
> flags.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/pull/7321#discussion_r1549678842>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A22KQEYY6YW5CWHE42JWMATY3P3Q7AVCNFSM6AAAAABFTWA2S2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNZWG4YTANJQGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
